vscode_extension/src/AacTaskProvider.ts

```diff
@@ -0,0 +1,124 @@
+            const definition: AacTaskDefinition = <any>_task.definition;
+            return new vscode.Task(definition, _task.scope ?? vscode.TaskScope.Workspace, definition.task, 'aac', new vscode.ShellExecution(`aac ${definition.task}`));
+        }
+        return undefined;
+    }
+}
+
+function execShell(command: string, options: cp.ExecOptions): Promise<{ stdout: string; stderr: string }> {
+    return new Promise<{ stdout: string; stderr: string }>((resolve, reject) => {
+        cp.exec(command, options, (error, stdout, stderr) => {
+            if (error) {
+                reject({ error, stdout, stderr });
+            }
+            resolve({ stdout, stderr });
+        });
+    });
+}
+
+let _channel: vscode.OutputChannel;
+function getOutputChannel(): vscode.OutputChannel {
+    // Creates a new entry in the "OUTPUT" panel at the bottom of the IDE.
+    if (!_channel) {
+        _channel = vscode.window.createOutputChannel('Architecture-as-code');
+    }
+    return _channel;
+}
+
+interface AacTaskDefinition extends vscode.TaskDefinition {
+    /**
+     * The task name
+     */
+    task: string;
+}
+
+async function execAacShellCommand(command: string, args: Array<string> = []): Promise<string> {
+
+    let result: string = "";
+
+    const outputChannel = getOutputChannel();
+
+    let commandArgsArray = ["aac", command, ...args];
+    try {
+        const { stdout, stderr } = await execShell(commandArgsArray.join(" "), {});
+        if (stderr && stderr.length > 0) {
+            outputChannel.appendLine(stderr);
+            outputChannel.show(true);
+        }
+        if (stdout) {
+            result = stdout;
+        }
+    } catch (error: any) {
+        let errorMessage = "";
+        if (error.stderr) {
+            errorMessage = error.stderr;
+        }
+        if (error.stdout) {
+            errorMessage = error.stderr;
+        }
+        outputChannel.appendLine(`Failed to execute AaC command:\n${errorMessage}`);
+        outputChannel.show(true);
+        throw error;
+    }
+
+    return result;
+}
+
+async function getAacTasks(): Promise<vscode.Task[]> {
+
+    return execAacShellCommand("-h").then(aacHelpOutput => {
+        const result: vscode.Task[] = [];
+
+        const regExp = /{(.*)}/;
+        const commandNamesMatch = regExp.exec(aacHelpOutput);
+
+        let commandNames: Array<string> = [];
+        if (commandNamesMatch && commandNamesMatch.length >= 2) {
+            commandNames = commandNamesMatch[1].split(",");
+        }
+
+        for (const commandName of commandNames) {
+
+            const commandNameRegexPattern = `\\\s+${commandName}[\\s|\\n]*\\s(.*)`;
+            const regex = new RegExp(commandNameRegexPattern, "g");
+            const matches = regex.exec(aacHelpOutput);
+            if (matches && matches.length > 0) {
+                // We can extract the command description with the following use of group matches
+                // const commandDescription = matches[1]
+
```

Comment:
```suggestion
```

Reply:
This will be necessary to define user input once we figure out how to get that to work with the extension.....